Add optional argument: -d

BHaFSec opened the pull request:
ex: python AWSBucketDump.py -l BucketNames.txt -g interesting_Keywords.txt -D -m 500000 -d 1

When -D is set, and -d is set to True or 1, create a directory for the given bucket hostname in which to save the files found in it.  If the directory already exists, use that existing directory.

This allows for a bit more sanity in storage locally when searching large numbers of buckets. :-)

BHaFSec committed:
Add new arg, -d

-d added to allow creation of individual directories for each host there were found files for, or if those directories already exist, use them.  This lets you keep things a bit more organized on the device running the script and storing results.

BHaFSec committed:
Merge pull request #1 from BHaFSec/BHaFSec-patch-1

Add new arg, -d

BHaFSec committed:
updating optional arguments list

Add -d specific text to the arguments list breakdown.

BHaFSec commented:
Note there's a change at line #162 also that is a comment mentioning optional grepList.  That was a change we were hoping to, but did not, get included in time to be in this pull request that would allow for downloading all found results when no grepList was defined.  Potentially something we finish later and submit in another pull.

jordanpotti commented on AWSBucketDump.py:
```diff
@@ -114,6 +120,20 @@ def downloadFiles(maxsize):
+                                                else:
+                                                        print("Using existing directory for host: (%s)" % local_savedir)
+
+                                                print("Saving file to: %s") % local_filename
```
`print("Saving file to: %s") % local_filename` should be `print("Saving file to: %s" % local_filename)`

BHaFSec replied:
Ah shoot. Apologies I missed this when doing brief testing last night.  'Tis what I get for not doing a full test after implementing a couple small changes at the last minute.  Thanks for catching it!

BHaFSec reviewed (commented):
Agreed!  Apologies I missed this before submitting the pull.  Thanks for catching it 👍

BHaFSec committed:
Fix print statement syntax in AWSBucketDump.py

Updated line 134 to include a fix for printing the file being saved to due to improper syntax.

jordanpotti commented:
Awesome, this was on my list of things to add!

jordanpotti commented:
Thanks @BHaFSec :)

BHaFSec commented:
Glad to help out, @jordanpotti :)
